index.js

```diff
@@ -1 +1,43 @@
-const app = "I don't do much."
+const products = [
+  { name: 'Head & Shoulders Shampoo', price: 5, discount: .6 },
+  { name: 'Twinkies', price: 8, discount: .45 },
+  { name: 'Oreos', price: 6.5, discount: .8 },
+  { name: 'Jasmine-scented bath pearls', price: 14, discount: .7 },
+];
+
+const couponLocations = [
+  { room: 'Living room', amount: 5 },
+  { room: 'Kitchen', amount: 2 },
+  { room: 'Bathroom', amount: 1 },
+  { room: 'Master bedroom', amount: 7 },
+];
+
+function reduce(collection, callback, initialValue) {
```

Comment:
I just went down a rabbit hole watching old functional (and "functional") programming conference talks tahnks to this function.

I think this is great, I just had to note that.